libraries/chain/contracts/evt_contract.cpp

```diff
@@ -760,15 +757,13 @@ apply_evt_executedelay(apply_context& context) {
         delay_def delay;
         tokendb.read_delay(edact.name, delay);
 
-        EVT_ASSERT(delay.signed_keys.find(edact.executor) != delay.signed_keys.end(), delay_executor_exception, "Executor hasn't sign his key on this delay transaction");
-
         auto now = context.control.head_block_time();
-        EVT_ASSERT(delay.status == delay_status::proposed, delay_status_exception, "Delay is not in 'proposed' status.");
+        EVT_ASSERT(delay.status == delay_status::proposed, delay_status_exception, "Delay is not in proper status.");
         EVT_ASSERT(delay.trx.expiration > now, delay_expired_tx_exception, "Delay transaction is expired at ${expir}, now is ${now}", ("expir",delay.trx.expiration)("now",now));
 
         auto strx = signed_transaction(delay.trx, delay.signatures);
         auto mtrx = std::make_shared<transaction_metadata>(strx);
-        auto trace = context.control.push_delay_transaction(mtrx, fc::time_point::maximum());
+        auto trace = context.control.push_delay_transaction(mtrx, now);
```

Comment:
Now is not valid here, because this parameter is the `deadline`. And it will be checked when the trx is prepared to execute. There's is a chance that at this second(aka. now) we push it, and it just executed in the next second, and it will be failed.
This is not excepted behavior, all the delay transactions should be accepted. And no need to set deadline. Because it just will be executed right now.